### PR TITLE
Make it possible to extract sub results from another results file

### PR DESCRIPTION
**Pull request by finsberg**
Adding this option so that we for example can extract only the last beat from a huge simualtion.

### CODE REVIEW

File: src/simcardems/postprocess.py [hunk 1]
```diff
@@ -553,3 +555,99 @@ def activation_map(
...
+    functions: Dict[str, Dict[str, dolfin.Function]] = defaultdict(dict)
+    for group_name, group in names.items():
+        for func_name in group:
+            try:
+                functions[group_name][func_name] = loader._functions[group_name][
+                    func_name
+                ]
+            except KeyError as e:
+                raise KeyError(
+                    "Invalid group {group_name} and function {func_name}",
```
Comment (jorgensd) [inline src/simcardems/postprocess.py:642]: Is this meant to be an f-string or is there some magic here?

File: tests/test_postprocessing.py [hunk 2]
```diff
@@ -41,3 +43,48 @@ def voltage_gen():
...
+def test_extract_sub_results(geo, tmp_path):
+    results_file = tmp_path / "results.h5"
+    collector = simcardems.DataCollector(
+        outdir=results_file.parent,
+        outfilename=results_file.name,
+        geo=geo,
+    )
+
+    # Setup a two mech functions and one ep function
+    V_mech = dolfin.FunctionSpace(geo.mesh, "CG", 1)
```
Comment (jorgensd) [inline tests/test_postprocessing.py:57]: "Lagrange" ?:D

File: src/simcardems/postprocess.py (hunk 1 above)
Comment (finsberg) [reply src/simcardems/postprocess.py]: Whoops, fixed in [95eb6b5](https://github.com/ComputationalPhysiology/simcardems/pull/175/commits/95eb6b5328e73dc9b38afc83b4b7467a79011839). Thanks

File: tests/test_postprocessing.py (hunk 2 above)
Comment (finsberg) [reply tests/test_postprocessing.py]: Sure [8cd601a](https://github.com/ComputationalPhysiology/simcardems/pull/175/commits/8cd601a756117fab0b62607234ea92a25604566b) 😎

**Review by jorgensd — APPROVED** [review]

**Comment by finsberg** [comment]
@IlsevanHerck Can we merge this?

**Pull request merged**